framework/lib/hooks/use-loading-message/index.ts

```diff
@@ -0,0 +1,15 @@
+import { useState } from 'react'
+import { useInterval } from '../use-interval'
+
+export const useLoadingMessage = (prefix: string = '') => {
+  const [ dots, setDots ] = useState('.')
+
+  useInterval(() => {
+    if (dots.length === 3) {
+      setDots('.')
+    } else {
+      setDots(`${dots}.`)
+    }
+  }, 300)
```

Comment:
this 300 should be  a parameter we can default it to 300 but we should be able to change it